tests/estimators/speech_recognition/test_tensorflow_lingvo.py

```diff
@@ -0,0 +1,329 @@
+                "utt_id",
+            }
+        )
+
+    def test_predict(self, audio_data):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = audio_data
+
+        lingvo = LingvoAsr()
+        predictions = lingvo.predict(test_input, batch_size=2)
+        assert predictions.shape[0] == test_input.shape[0]
+        assert isinstance(predictions[0], np.str_)
+
+    def test_loss_gradient_tensor(self, audio_batch_padded):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input, test_mask_frequency = audio_batch_padded
+        test_target_dummy = np.array(["DUMMY"] * test_input.shape[0])
+
+        lingvo = LingvoAsr()
+        feed_dict = {
+            lingvo._x_padded: test_input,
+            lingvo._y_target: test_target_dummy,
+            lingvo._mask_frequency: test_mask_frequency,
+        }
+        loss_gradient = lingvo._sess.run(lingvo._loss_gradient_op, feed_dict)
+        assert test_input.shape == loss_gradient.shape
+        assert loss_gradient.sum() == 0.0
+
+    def test_loss_gradient_per_batch(self, audio_data):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = audio_data
+        test_target = np.array(["This", "is", "a dummy", "a dummy"])
+
+        lingvo = LingvoAsr()
+
+        gradients = lingvo._loss_gradient_per_batch(test_input, test_target)
+        gradients_abs_sum = np.array([np.abs(g).sum() for g in gradients], dtype=object)
+
+        # test shape, equal inputs have equal gradients, non-zero inputs have non-zero gradient sums
+        assert [x.shape for x in test_input] == [g.shape for g in gradients]
+        assert_allclose(np.abs(gradients[2]).sum(), np.abs(gradients[3]).sum(), rtol=1e-01)
+        assert_array_equal(gradients_abs_sum > 0, [False, True, True, True])
+
+    def test_loss_gradient_per_sequence(self, audio_data):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = audio_data
+        test_target = np.array(["This", "is", "a dummy", "a dummy"])
+
+        lingvo = LingvoAsr()
+
+        gradients = lingvo._loss_gradient_per_sequence(test_input, test_target)
+        gradients_abs_sum = np.array([np.abs(g).sum() for g in gradients], dtype=object)
+
+        # test shape, equal inputs have equal gradients, non-zero inputs have non-zero gradient sums
+        assert [x.shape for x in test_input] == [g.shape for g in gradients]
+        assert_allclose(np.abs(gradients[2]).sum(), np.abs(gradients[3]).sum(), rtol=1e-01)
+        assert_array_equal(gradients_abs_sum > 0, [False, True, True, True])
+
+
+@pytest.mark.skipif(tf.__version__ != "2.1.0", reason="requires Tensorflow 2.1.0")
+@pytest.mark.skipMlFramework("pytorch", "mxnet", "kerastf", "non_dl_frameworks")
+class TestLingvoAsrLibriSpeechSamples:
+    # specify LibriSpeech samples for download and with transcriptions
+    samples = {
```

Comment:
With the new test framework, we have a new nice functionality which enables us to hide long arrays or json from the test code to make it more readable. This is a perfect use case with the variable `samples`. By using the `store_expected_values` fixture to store that json object and then the `expected_values` fixture in each test to retrieve it, this will automatically create a separate json file for this test script which each test will leverage. There's more detailed explanation in section 6 of the [wiki](https://github.com/Trusted-AI/adversarial-robustness-toolbox/wiki/ART-Unit-Testing). It really just takes a few seconds to do but if anything is unclear, I'm more than happy to help you of course :-)

Reply:
Good point. Maybe we should go even one step further and add the three Librispeech samples as fixtures to the tests, e.g. similar as we do for images with MNIST data. That would also make your previous comment redundant regarding the duplicate code for loading the samples. 

I would suggest to add this as a seperate feature request for the near future.

Reply:
yes I agree that would be very good indeed, as I'm sure we will be reusing these audio files as ART further develops its speech functionalities. Also it would make the test more robust as it would not be dependent on downloading this file for each test. @beat-buesser what's the procedure for this, can Hessel create a new issue for this so that we can track the need for this feature?

Reply:
I have created an issue for adding Librispeech sample fixtures :+1: